### PR TITLE
INTYGFV-16084: Added GlobalExceptionHandlerController that handles communication errors for integrated services.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**

### CODE REVIEW

**Comment by ChrilleSigerhed** [comment]
Regarding this function:
```java
       .onErrorMap(
            WebClientRequestException.class,
            ExceptionThrowableFunction.get(APPLICATION_WEBCERT)
        )
```
This will map any exception of type WebClientRequestException to our custom Exception called IntegrationServiceException. This exception only occurs if we are not able to establish communication with the service we are looking for.

If we want to throw this exception even if a 500-error occurs in the service we are calling, some modification will be needed.

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Looks good! But I have a few thoughts that we should consider:
1) If the call to the integrated service receives timeout we should also return a 503.
2) We need to log the stack trace when logging the error.
3) Need to include the original exception in the IntegrationServiceException.
4) Also add this to the intyg-proxy-service integration

File: integration-intyg-proxy-service/src/main/java/se/inera/intyg/minaintyg/integration/intygproxyservice/person/client/GetPersonFromIntygProxyServiceImpl.java [hunk 1]
```diff
@@ -1,13 +1,18 @@
 package se.inera.intyg.minaintyg.integration.intygproxyservice.person.client;
 
+import static se.inera.intyg.minaintyg.integration.common.constants.ApplicationConstants.APPLICATION_WEBCERT;
```
Comment (mww-sopra-steria) [inline integration-intyg-proxy-service/src/main/java/se/inera/intyg/minaintyg/integration/intygproxyservice/person/client/GetPersonFromIntygProxyServiceImpl.java:3]: Should use an application constant for INTYG_PROXY_SERVICE

**Review by mww-sopra-steria — APPROVED** [review]
Well done! Just change the constant for intyg-proxy-service

**Pull request merged**